### PR TITLE
Fix for issue 1174

### PR DESCRIPTION
**Pull request by walsh2000**
This avoids the problems mentioned in #1174 by removing our target on the Interactive Pop Gesture at viewDidDisappear-time
It avoid regressing to the problems documented in #257 by capturing the UIGestureRecognizer (as weak) at the time we addTarget. This allow us to remember on whom we should removeTarget.

I will have a separate commit to add a 2-level navigation-controller push to the Demo project (PR #1176), but that is not directly related to #1174 nor #257

### CODE REVIEW

File: JSQMessagesViewController/Controllers/JSQMessagesViewController.m [hunk 1]
```diff
@@ -1038,14 +1040,15 @@ - (void)jsq_registerForNotifications:(BOOL)registerForNotifications
...
 {
-    if (self.navigationController.interactivePopGestureRecognizer) {
-        [self.navigationController.interactivePopGestureRecognizer removeTarget:nil
-                                                                         action:@selector(jsq_handleInteractivePopGestureRecognizer:)];
-        
-        if (addAction) {
-            [self.navigationController.interactivePopGestureRecognizer addTarget:self
-                                                                          action:@selector(jsq_handleInteractivePopGestureRecognizer:)];
-        }
+    if (_currentInteractivePopGestureRecognizer) {
```
Comment (jessesquires) [inline JSQMessagesViewController/Controllers/JSQMessagesViewController.m]: prefer property access, `self.currentInteractivePopGestureRecognizer`

**Comment by jessesquires** [comment]
Awesome. Thanks @walsh2000 !

**Comment by walsh2000** [comment]
@jessesquires 

I was sure I saw a message from you asking that the property accesses go through accessors, but I don't see that comment anywhere anymore, but I made the changes.

Note, that I was following what appeared to be the pattern for "_jsq_isObserving" (see jsq_removeObservers)

**Comment by walsh2000** [comment]
Also, if I caused this Travis failure, I am not sure how. :smiling_imp:

File: JSQMessagesViewController/Controllers/JSQMessagesViewController.m (hunk 1 above)
Comment (walsh2000) [inline JSQMessagesViewController/Controllers/JSQMessagesViewController.m]: found your comment (finally)

**Comment by jessesquires** [comment]
@walsh2000 - no worries. looks like an issue with `xctool`. I just changed the script to use `xcodebuild`. if you sync with `develop` it should be ok

**Comment by walsh2000** [comment]
@jessesquires 

Out of curiosity, what is the timetable for 7.1.2?

**Comment by jessesquires** [comment]
@walsh2000 - lol -- ASAP! I wanted to do it last weekend, but didn't have a chance. Hopefully this week. :smile:

**Comment by jessesquires** [comment]
Hey @walsh2000 - there was one regression here. see 875eab6.

The pop gesture *must* be removed in `viewWillDisappear:`. Otherwise, the following will happen:

1. Drag to back
2. Stop ~half-way to allow VC to snap back (i.e., you stay on the message view, you don't pop the VC)
3. The keyboard will remain up and the toolbar will hide behind it

Please let me know if 875eab6 causes problems. :smile:

**Comment by jessesquires** [comment]
Oh well, after testing with #1176 there's no crash, so... :tada: I guess we're good?

**Comment by walsh2000** [comment]
@jessesquires 

Regarding 875eab6 & the regression.
I've stepped through the regression.
The app gets a series of keyboard frame changes (hiding then reappearing).
In the didHide we set toolbarBottomLayoutGuide to 0.
The subsequent keyboardDidChangeFrame has the right rectangle to place the inputToolbar in the right spot. However, we do not process the message before the inputToolbar is not first responder
````
    if (![self.inputToolbar.contentView.textView isFirstResponder] && self.toolbarBottomLayoutGuide.constant == 0.0f) {
        return;
    }
````

If we make the inputToolbar's text view the first responder on Cancelled/Ended, then the toolbar does not get lost behind the keyboard.

**Comment by walsh2000** [comment]
875eab6 effectively reverts #1175, so I found a solution for both #1174 & which does not introduce the inputToolbar regression. PR #1189

**Pull request merged**